src/gui/sdlmain.cpp

```diff
@@ -198,6 +198,11 @@ static DosBox::Rect to_rect(const SDL_Rect r)
 	return {r.x, r.y, r.w, r.h};
 }
 
+static SDL_Rect to_sdl_rect(const DosBox::Rect r)
```

Comment:
For small things, references can actually make things worse. Small types can be passed in registers, there's no forced memory indirection, etc.

I looked this up a while ago and structs under about 16-byte in size are better to passed by value. Makes perfect sense with 128-bit wide registers being standard. E.g. check this answer:

https://stackoverflow.com/questions/3314010/should-small-simple-structs-be-passed-by-const-reference

SDL_Rect can be 256-bits, so if we're in micro optimising territory, passing 256-bits by copy is potentially a lot cheaper than memory indirection.

Reply:
Wasn't meaning to micro-optimize.. just the old rule of thumb to use `const ref` for compound types (more a consistency things).

I guess setting thresholds when to use references versus copies is more of a micro-optimization thing. Now that we've silenced PVS Studio on these, I guess it won't be complaining about them :-)

Reply:
Ok, I can do it for consistency as it doesn't actually matter much, yeah 😄 

Well, I'd prefer to always write `const Type` and never think about it... I actually don't get why can't the compiler figure that out... hey, it's a const, so you pass it by value or reference, whatever is better, but we're not gonna mutate it.

I guess it just annoys me to know I'm passing something by reference when I know it can fit into a register 😄

Reply:
@kcgen Related to this, PVS Studio started complaining about this:

V835. Passing cheap-to-copy argument by reference may lead to decreased performance.
https://pvs-studio.com/en/docs/warnings/v835/

So that's basically what I was talking about 😄 

This pass by reference triggered it

```cpp
static DosBox::Rect to_rect(const SDL_Rect& r)
```

I'm passing `DosBox::Rect` by ref everywhere now for consistency, and it doesn't complain, so I'll leave those. I guess I just wanted to show that it has some heuristics for raising warnings for inefficient pass by refs, which is cool.